aten/src/ATen/core/ivalue.h

```diff
@@ -166,6 +204,22 @@ struct CAFFE2_API IValue final {
     std::swap(tag, rhs.tag);
   }
 
+
```

Comment:
Let's avoid declaring equality for IValues generally and instead have a definition used by Dict objects specifically.  The problem is that equality for these values is poorly defined (is it python ==, is, or something else?). It is better to just ask how we want to dict to behave.